src/python/grpcio/grpc/__init__.py

```diff
@@ -1021,6 +1021,105 @@ def set_details(self, details):
         raise NotImplementedError()
 
 
+##############  Service-Side Interceptor Interfaces & Classes  #################
+
+
+class UnaryUnaryServerInterceptor(six.with_metaclass(abc.ABCMeta)):
+    """Affords intercepting unary-unary RPCs on the service-side.
+
+This is an EXPERIMENTAL API and is subject to change without notice."""
+
+    @abc.abstractmethod
+    def intercept_unary_unary_handler(self, handler, method, request,
+                                      servicer_context):
+        """Intercepts unary-unary RPCs on the service-side.
+
+    Args:
+      handler: The handler to continue processing the RPC.
+      It takes a request value and a ServicerContext object
+      and returns a response value.
+      method: The full method name of the RPC.
+      request: The request value for the RPC.
+      servicer_context: The context of the current RPC.
```

Comment:
I have been playing with the current implementation. Is `servicer_context` supposed to be an instance of `grpc._server._Context`?

Reply:
`servicer_context` is supposed to be an instance of `grpc.ServicerContext`. We wouldn't expose in our API (even an experimental API) an internal class like `grpc._server._Context`.

Why isn't the `method` parameter passed to this method the right way to get the method name?

Reply:
Sorry I originally missed getting back to this comment notification. Definitely we don't want exposure of the private API. I suspect the issue here might be that the method name would be full HTTP path e.g. `/Greeter/Hello` and the desired part would be `Hello`? The reality is the gRPC server object does not want to couple itself with any specific form of handler, and a `GenericRpcHandler` can choose to service the RPC however it wants. In other words, there is no need for a one to one mapping to exist between RPC method names (as given to the interceptor) and methods on the Python servicer object. I think the best thing would be to parse the method name out of full method name string.